apiv2/pkg/clients/robot/robot.go

```diff
@@ -87,14 +87,31 @@ func (in AccessAction) String() string {
 
 // ListRobotAccounts ListProjectRobots returns a list of all robot accounts.
 func (c *RESTClient) ListRobotAccounts(ctx context.Context) ([]*modelv2.Robot, error) {
-	resp, err := c.V2Client.Robot.ListRobot(&robot.ListRobotParams{
-		Context: ctx,
-	}, c.AuthInfo)
-	if err != nil {
-		return nil, handleSwaggerRobotErrors(err)
+	var robotAccounts []*modelv2.Robot
+	var page int64 = c.Options.Page
+	var pageSize int64 = c.Options.PageSize
```

Comment:
When instantiating a new client e.g. via [`NewRESTClient()`](https://github.com/mittwald/goharbor-client/blob/master/apiv2/client.go#L88) you can pass an optional [`*config.Options`](https://github.com/mittwald/goharbor-client/blob/master/apiv2/pkg/config/options.go#L7). By this, the values for `page` or `pageSize` should be implicitly set when calling `ListRobotAccounts()`. See below comment for an example.

Reply:
But in [GetRobotAccountByName()](https://github.com/mittwald/goharbor-client/blob/master/apiv2/pkg/clients/robot/robot.go#L103)/ [DeleteRobotAccountByName](https://github.com/mittwald/goharbor-client/blob/master/apiv2/pkg/clients/robot/robot.go#L146)/ [RefreshRobotAccountSecretByName](https://github.com/mittwald/goharbor-client/blob/master/apiv2/pkg/clients/robot/robot.go#L145) need to call `ListRobotAccounts()`. If *total robot accounts > pageSize* (which is enforced by the Goharbor API to maximum of 100) and *the matched robot account is >= .no 101*, `ListRobotAccounts()` couldn't list all, so above calling methods will return `errors.ErrRobotAccountUnknownResource`

Reply:
Thanks for pointing that out, @manhtukhang! I did not have in mind that the Goharbor API limits the maximum `PageSize` to 100 items (per request). Your solution might then also be applicable to other `List[...]` methods in this client. Anyhow, the call to `c.V2Client.Robot.ListRobot` should ideally contain the fields from [`*config.Options`](https://github.com/mittwald/goharbor-client/blob/master/apiv2/pkg/config/options.go#L7). If you could add those to your PR, i think it'll be ready to go! :+1: